shotover-proxy/src/transforms/mod.rs

```diff
@@ -280,6 +280,8 @@ pub struct Wrapper<'a> {
     transforms: Vec<&'a mut Transforms>,
     pub client_details: String,
     chain_name: String,
+    /// When true transforms must flush any buffered messages into the messages field
```

Comment:
Let's expand the documentation here to explain why (e.g. a flush can be set to true on shutdown, or auth context changes)

Reply:
"auth context change" doesnt sound quite right to me.
Does what I wrote instead make sense?

Reply:
If I change the user I'm authenticated as in the middle of the session. Then we should probably flush all previous messages sent under the old auth context

Reply:
oh I see.
If its desirable to set flush at any time for protocol reasons then maybe we shouldnt try to exhaustively document the reasons.
What about:
`This can occur at any time but will always occur before the transform is destroyed due to either shotover or the transforms chain shutting down.`

Reply:
sounds good